src/grav/test.py

```diff
@@ -0,0 +1,115 @@
+"""
+This file contains meat of the Gravitational QA coding challenge.
+
+The test() function performs the test as requested in the challenge.
+
+The setup() function does all the provisioning necessary to go from:
+  "package is installed" -> "start/stop/logs a wellknown container"
+
+The teardown() function attempts to unwind any dangling resources
+from setup() and test(). Ideally leaving a clean test enviromnent
+if it runs to completion.  This is important for idempotency.
+"""
+import re
+from pkg_resources import resource_filename
+
+from grav.client import demux_logs, DockerClient
+
+
+def setup():
+    # If this were real, I'd pass these magic strings stattered through
+    # file as params and have them loaded from a config.
+    image_tar = resource_filename(__name__, "wellknown.tar")
+
+    # It'd be better to nest client, image, container setup/teardown
+    # so that in the case client & image succeed but container fails
+    # we only run the teardown for image & client, and skip container
+    # teardown since it was never set up.
+    #
+    # That is more effort than this sample merits imo, so setup/teardown
+    # are all in one. -- wdella 2019-10
+    client = DockerClient("/var/run/docker.sock")
+
+    # load image from tar, get image id
+    with open(image_tar, 'rb') as fh:
+        data = fh.read()
+
+    headers = {'Content-Type': 'application/tar', 'Content-Length': len(data)}
+    resp = client.post("/images/load?fromSrc=-", body=data, headers=headers)
```

Comment:
nit: I would suggest to use urllib to create paths

Reply:
I see no advantage to urllib in a project of this scope, and it is more verbose than string formatting/concatenation.  Certainly in a larger project it is my go-to for avoiding url concatenation/formatting errors.

Let me know if you consider this a blocker - otherwise I do not intend to change it.